.github/workflows/docker.yaml

```diff
@@ -0,0 +1,810 @@
+---
+
+name: 'Docker'
+
+
+on:
+  workflow_call:
+    inputs:
+      DOCKER_BUILD_REGISTRY:
+        default: "ghcr.io"
+        description: Registry where build image will live
+        required: false
+        type: string
+      DOCKER_BUILD_IMAGE_NAME:
+        default: "${{ github.repository }}"
+        description: Docker image name for building image
+        required: false
+        type: string
+      DOCKER_BUILD_IMAGE_TAG:
+        default: "${{ github.sha }}"
+        description: Docker image tag for building image
+        required: false
+        type: string
+      DOCKER_PUBLISH_REGISTRY:
+        default: "docker.io"
+        description: Registry where image will be published to
+        required: false
+        type: string
+      DOCKER_PUBLISH_IMAGE_NAME:
+        default: "${{ github.repository }}"
+        description: Docker image name for publishing the image
+        required: false
+        type: string
+      DOCKER_SCAN_IMAGE_VULNERABILITY:
+        default: true
+        description: Scan Image with Trivy
+        required: false
+        type: boolean
+      DOCKER_TAG_IMAGE_TAG_SOURCE:
+        default: "${{ github.sha }}"
+        description: Docker image tag Used for source Image to tag
+        required: false
+        type: string
+      DOCKER_TAG_IMAGE_TAG_LATEST:
+        default: true
+        description: Create image with tag 'Latest'
+        required: false
+        type: boolean
+      DOCKER_TAG_IMAGE_TAG_SHA:
+        default: true
+        description: Create image with tag that is the git commit
+        required: false
+        type: boolean
+
+
+permissions:
+  pull-requests: write
+  contents: write
+  actions: read
+  checks: write
+  packages: write
+  security-events: write
+  statuses: write
+  id-token: write
+  attestations: write
+
+jobs:
+
+  docker-build:
+    if:
+      (${{
+        github.event.push
+          ||
+        github.ref_type == 'tag'
+      }})
+    runs-on: ubuntu-latest
+    name: Build Image
+    outputs:
+      build-image-digest: ${{ steps.image-digest.outputs.value-out }}
+    steps:
+
+
+      - uses: actions/checkout@v4
+
+      - name: Dummy Task
+        if: ${{ github.ref_type == 'tag' }}
+        run: |
+          echo "This Job does nothing however as 'docker-build' is a dependency job, must do something.";
+
+
+
+      - name: Log into registry ghcr.io
+        uses: docker/login-action@343f7c4344506bcbf9b4de18042ae17996df046d # v3.0.0
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+
+      - name: Setup BuildX
+        if: ${{ github.ref_type != 'tag' }}
+        run: |
+          docker buildx create --name project-v3-builder;
+          docker buildx use project-v3-builder;
+
+
+      - name: build image
```

Comment:
# Fix

the tag on a forked repo needs to go to the forked repo's docker registry, not this repo's registery.